### PR TITLE
Fix Boolean rowsList or colsList output

### LINKED ISSUE #10658: Boolean rowsList or colsList return incorrect Matrix
```
>>> m = Matrix(3, 3, range(9))
>>> m.extract([0, 1, 2], [False, True, False])
⎡ 0 1 0 ⎤
| 3 4 3 |
⎣ 6 7 6 ⎦
```
it's return wrong output.when we takes ```boolean``` value for ```rowsList``` or ```colsList```.

### PR DESCRIPTION
**Pull request by mvnnn**
fixes #10658

### CODE REVIEW

**Comment by mvnnn** [comment]
ping @asmeurer

File: sympy/matrices/matrices.py [hunk 1]
```diff
@@ -1053,9 +1053,26 @@ def extract(self, rowsList, colsList):
         [2],
         [8]])
 
+        when colsList or rowsList is boolean then indexing will start at 0.
```
Comment (asmeurer) [inline sympy/matrices/matrices.py]: You should put a description of what it does when there are boolean values here. Something like "rowsList or colsList can also be a list of booleans, in which case the rows or columns corresponding to the True values will be selected".

File: sympy/matrices/matrices.py [hunk 2]
```diff
@@ -1053,9 +1053,26 @@ def extract(self, rowsList, colsList):
...
+        [6,  8],
+        [9, 11]])
         """
+
         cols = self.cols
         flat_list = self._mat
+        if rowsList and all(isinstance(i, bool) for i in rowsList):
+            rowsList = [index for index, item in enumerate(rowsList) if item]
+            if not rowsList:
+                return
```
Comment (asmeurer) [inline sympy/matrices/matrices.py]: Should return an empty matrix (with appropriate shape) here.

Comment (mvnnn) [inline sympy/matrices/matrices.py]: No, that's not return empty matrix when we use boolean value of rowList or colsList. if we need that then i will remove 
 ```
if not rowsList:
      return
```
that condition then it's return empty Matrix.

Comment (mvnnn) [inline sympy/matrices/matrices.py]:
```
>>> m.extract([0, 1, 3], [])
Matrix(3, 0, [])
```
@asmeurer we need output ```Matrix([])``` here.

**Comment by mvnnn** [comment]
ping @asmeurer

**Comment by mvnnn** [comment]
@asmeurer now, that's looks good.

**Comment by mvnnn** [comment]
```
Should return an empty matrix (with appropriate shape)
```
@asmeurer yes. i added test cases for this. now, it's looks good ?

**Comment by mvnnn** [comment]
@asmeurer @smichr now, it's looks good ?

**Comment by mvnnn** [comment]
@asmeurer now, it's looks good ?

**Comment by mvnnn** [comment]
@asmeurer now, it's looks good ?

**Pull request merged**